src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SearchController.kt

```diff
@@ -1,34 +1,116 @@
 package uk.gov.justice.digital.hmpps.personrecord.api.controller
 
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.Parameter
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import io.swagger.v3.oas.annotations.tags.Tag
 import jakarta.validation.constraints.NotBlank
 import org.springframework.security.access.prepost.PreAuthorize
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.personrecord.api.constants.Roles
 import uk.gov.justice.digital.hmpps.personrecord.api.controller.exceptions.PersonRecordNotFoundException
 import uk.gov.justice.digital.hmpps.personrecord.api.model.PersonIdentifierRecord
 import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonEntity
 import uk.gov.justice.digital.hmpps.personrecord.jpa.repository.PersonRepository
 import uk.gov.justice.digital.hmpps.personrecord.model.types.SourceSystemType
+import uk.gov.justice.hmpps.kotlin.common.ErrorResponse
 
+@Tag(name = "Search")
 @RestController
-@PreAuthorize("hasRole('ROLE_CORE_PERSON_RECORD_API__SEARCH__RO')")
+@PreAuthorize("hasRole('${Roles.SEARCH_API_READ_ONLY}')")
 class SearchController(
   private val personRepository: PersonRepository,
 ) {
 
+  @Operation(description = "Search for person record and associated records with a CRN within the system")
   @GetMapping("/search/offender/{crn}")
-  fun searchByCrn(@NotBlank @PathVariable(name = "crn") crn: String): List<PersonIdentifierRecord> {
+  @ApiResponses(
+    ApiResponse(responseCode = "200", description = "OK"),
+    ApiResponse(
+      responseCode = "400",
+      description = "Invalid request.",
+      content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+    ),
+    ApiResponse(
+      responseCode = "404",
+      description = "Requested resource not found.",
+      content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+    ),
+    ApiResponse(
+      responseCode = "500",
+      description = "Unrecoverable error occurred whilst processing request.",
+      content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+    ),
+  )
+  fun searchByCrn(
+    @NotBlank
+    @PathVariable(name = "crn")
+    @Parameter(description = "The identifier of the probation source system (nDelius)", required = true)
+    crn: String,
+  ): List<PersonIdentifierRecord> {
     return handlePersonRecord(personRepository.findByCrn(crn), crn)
   }
 
+  @Operation(description = "Search for person record and associated records with a prisoner number within the system")
```

Comment:
prison number or prisoner number? We have both here. I am sure we have picked one in the codebase, let's go with that throughout

Reply:
from the code below, it looks like prisonNumber is the one we picked